src/charm.py

```diff
@@ -43,7 +46,16 @@ def __init__(self, *args: Any) -> None:
         except CharmConfigInvalidError as exc:
             self.model.unit.status = ops.BlockedStatus(exc.msg)
             return
-        self._synapse = Synapse(charm_state=self._charm_state)
+        self._database = DatabaseObserver(self)
+        self.framework.observe(
+            self._database.database.on.database_created, self._on_database_created
+        )
+        self.framework.observe(
+            self._database.database.on.endpoints_changed, self._on_config_changed
```

Comment:
The guidance is to have one event handler per event with a name that matches the event. Events can share functionality from within those event handlers by calling underlying functions

Reply:
My bad, changed to use same name as the event.